src/server/qtquick/woutputrenderwindow.cpp

```diff
@@ -1111,9 +1111,13 @@ bool OutputHelper::tryToHardwareCursor(const LayerData *layer)
             m_cursorLayerProxy->setHeight(buffer->height);
 
             auto newBuffer = m_cursorRenderer->lastBuffer();
-            const QSize newBufferSize(newBuffer->handle()->width,
-                                      newBuffer->handle()->height);
-            if (m_cursorDirty || !newBuffer || pixelSize != newBufferSize) {
+
+            // When the cursor moves to the edge of the screen, it will be truncated by default,
+            // and the part beyond the screen will not be included in the pixelSize. In this case,
+            // the pixelSize is inconsistent with the newBuffer size, and it needs to be redrawn.
+            if (m_cursorDirty || !newBuffer || (newBuffer &&
```

Comment:
Not need `newBuffer &&`, because it has `!newBuffer`.

Reply:
done